core/src/tile/labels/labels.cpp

```diff
@@ -32,7 +36,13 @@ bool Labels::addLabel(MapTile& _tile, const std::string& _styleName, Label::Tran
             return false;
         }
 
-        l->update(m_view->getViewProjectionMatrix() * _tile.getModelMatrix(), m_screenSize, 0);
+        // NB: viewOrigin.z is only determined by screen width and height.
+        const auto& viewOrigin = m_view->getPosition();
+
+        auto modelMatrix = glm::scale(glm::mat4(1.0), glm::vec3(_tile.getScale()));
+        modelMatrix[3][2] = -viewOrigin.z;
+
+        l->update(m_view->getViewProjectionMatrix() * modelMatrix, m_screenSize, 0);
```

Comment:
This is how the modelmatrix of a tile is calculated by the call tile->update(view) in tileworker, minus the use of the current view position. 
I wanted to remove the access the current view state from a worker thread. So this (should) set up the labels as if the tile was at the map center.